### ISSUE TITLE
Default trait methods trigger `unused_async`

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#11660]
Empty default methods shouldn't trigger `unused_async` because downstream users can overwrite and make use of `async` functionalities.

```rust
#![allow(async_fn_in_trait)]
#![deny(clippy::unused_async)]
#![feature(async_fn_in_trait)]

pub trait Foo {
    async fn bar(&mut self) {}
}
```

**Comment by c410-f3r** [comment]
On a second thought, the lint is acting as intended. I guess ppl should use `async_fn_in_trait` for this particular case.

**Comment by oxalica** [comment]
> On a second thought, the lint is acting as intended. I guess ppl should use `#[allow(..)]` for this particular case.

Could you give some explanations why we *should* warn this case? I also encountered this because a public trait is for asynchronous event handling and the default impl is a no-op.

Even if we do want to lint for this case, it should be at least gated under [`avoid-breaking-exported-api = false`](https://doc.rust-lang.org/nightly/clippy/lint_configuration.html#avoid-breaking-exported-api), while is not the case currently.

**Comment by konstin** [comment]
Sorry for reviving this, but i have the exact same problem, the code below triggers this lint, even though it can't write this method as a sync function, and since the trait is private, it's detectable that there is an async override:

```rust
trait AsyncDirectoryWrite: Sized {
    /// ...

    async fn close(self) -> Result<(), Error> {
        Ok(())
    }
}

struct AsyncZipWriter(ZipFileWriter<tokio_util::compat::Compat<File>>);

impl AsyncDirectoryWrite for AsyncZipWriter {
    /// ...

    async fn close(self) -> Result<(), Error> {
        self.0.close().await?;
        Ok(())
    }
}
```
